### PR TITLE
SNOW-488437 Add connection timeout

### PR DESCRIPTION
**Pull request by sfc-gh-anwang**
Resolves issues related to this [ticket](https://snowflakecomputing.atlassian.net/browse/SNOW-488437). 
Currently, the Java SDK does not timeout on HTTP connections, I added timeout in the RequestConfig.

### CODE REVIEW

File: src/main/java/net/snowflake/ingest/utils/HttpUtil.java [hunk 1]
```diff
@@ -42,7 +43,7 @@
 
   private static final String PROXY_SCHEME = "http";
   private static final int MAX_RETRIES = 3;
-
+  private static final int TIMEOUT = 300; // 5 minutes
```
Comment (sfc-gh-japatel) [inline src/main/java/net/snowflake/ingest/utils/HttpUtil.java:46]: nit: CONNECTION_TIMEOUT

File: src/main/java/net/snowflake/ingest/utils/HttpUtil.java [hunk 2]
```diff
@@ -64,7 +65,16 @@ private static void initHttpClient() {
...
         new SSLConnectionSocketFactory(
             sslContext, new String[] {"TLSv1.2"}, null, new DefaultHostnameVerifier());
-
+    // Set connectionTimeout which is the timeout until a connection with the server is established
+    // Set connectionRequestTimeout which is the time to wait for getting a connection from the
+    // connection pool
+    // Set socketTimeout which is the max time gap between two consecutive data packets
+    RequestConfig requestConfig =
+        RequestConfig.custom()
+            .setConnectTimeout(TIMEOUT * 1000)
```
Comment (sfc-gh-japatel) [inline src/main/java/net/snowflake/ingest/utils/HttpUtil.java:74]: Can you use 
```TimeUnit.SECONDS.convert(CONNECTION_TIMEOUT_MINS, TimeUnit.MINUTES)```

File: src/main/java/net/snowflake/ingest/utils/HttpUtil.java (hunk 1 above)
Comment (sfc-gh-japatel) [reply src/main/java/net/snowflake/ingest/utils/HttpUtil.java:46]: Is 5 minutes too long a time? What does JDBC use?

Comment (sfc-gh-anwang) [reply src/main/java/net/snowflake/ingest/utils/HttpUtil.java:46]: they use 5 mins for Socket timeout, and the other two are 1 min

Comment (sfc-gh-japatel) [reply src/main/java/net/snowflake/ingest/utils/HttpUtil.java:46]: I think 5 mins is too long a time for waiting. 
Also, https://github.com/snowflakedb/snowflake-jdbc/blob/69f560b9370cbe5de112f259600911ccd74333bf/src/main/java/net/snowflake/client/jdbc/telemetryOOB/TelemetryService.java

Comment (sfc-gh-anwang) [reply src/main/java/net/snowflake/ingest/utils/HttpUtil.java:46]: They also have the timeout in the util, https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/core/HttpUtil.java

Comment (sfc-gh-japatel) [reply src/main/java/net/snowflake/ingest/utils/HttpUtil.java:46]: Got it! Thanks! I think it is okay for us to start with this timeout. 
CC: @sfc-gh-rramachandran

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm!

**Pull request merged**